### ISSUE TITLE
[🐞] Loader state lost when setting local state

### ISSUE THREAD

**Issue by rolanday** [BuilderIO/qwik#3308]
### Which component is affected?

Qwik City (routing)

### Describe the bug

Issue:
Loader state lost or cleared from view when setting local signal or store value.

Qwik version info:
"@builder.io/qwik": "0.21.0",
"@builder.io/qwik-city": "0.5.3",

### Reproduction

https://github.com/rolanday/qwik-loader

### Steps to reproduce

1. Create a new project using npm create qwik@latest
2. Edit `src/routes/index.tsx` as shown below.
3. `npm run dev`
4. Click the button and observe rendered view

```
// src/routes/index.tsx
import { component$, useStore } from "@builder.io/qwik";
import type { DocumentHead } from "@builder.io/qwik-city";
import { routeLoader$ } from "@builder.io/qwik-city";

export const useLoaderData = routeLoader$(() => {
  return { name: "malmsteen" };
});

export default component$(() => {
  const loaderData = useLoaderData();
  const localStore = useStore({ x: 1 }); // problem also occurs if this is a signal instead of store
  return (
    <div>
      <pre>localStore: {JSON.stringify(localStore, null, 2)}</pre>
      <pre>loaderData: {JSON.stringify(loaderData.value, null, 2)}</pre>
      <button
        onClick$={() => {
          // console.log(JSON.stringify(loaderData.value)); <== Uncommenting this line solve the problem, but why?
          localStore.x = 2;
        }}
      >
        {localStore.x}
      </button>
    </div>
  );
});

export const head: DocumentHead = {
  title: "Welcome to Qwik",
  meta: [
    {
      name: "description",
      content: "Qwik site description",
    },
  ],
};

```

### System Info

```shell
System:
    OS: macOS 12.6
    CPU: (48) x64 Intel(R) Xeon(R) W-3265M CPU @ 2.70GHz
    Memory: 82.30 GB / 192.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.16.0 - /usr/local/bin/node
    Yarn: 1.22.15 - /usr/local/bin/yarn
    npm: 8.11.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 110.0.5481.177
    Firefox Developer Edition: 111.0
    Safari: 16.0
  npmPackages:
    @builder.io/qwik: 0.21.0 => 0.21.0
    @builder.io/qwik-city: 0.5.3 => 0.5.3
    undici: 5.20.0 => 5.20.0
    vite: 4.1.4 => 4.1.4
```

### Additional Information

Issue also reproduced by other Discord member: https://discord.com/channels/842438759945601056/1082965605054955561/1083096120722464778

Video link showing the problem in action:

https://user-images.githubusercontent.com/3402123/223807825-fd4dd82a-cff6-4155-ac52-fb4c330812ca.mov

**Comment by jihoon416** [comment]
Not sure if it is caused by the same problem, but I also had problems with loader data being absent. (I'm quite new to Qwik so there's also a possibility I might be using it wrong, please correct me if I am wrong)

I started a new project, and in the `src/routes/todolist/index.tsx` file, I just added a `<span>{`${action.isRunning}`}</span>` and an error like below popped up below.

![image](https://user-images.githubusercontent.com/63445490/224068018-d15063f7-b570-4c50-af73-65b0da0999c5.png)

To reproduce, first add some items by submitting the form, then reload the page and try adding a item by submitting the form.

https://user-images.githubusercontent.com/63445490/224070773-dffcf0ea-5473-4eed-af8d-e5195ac35245.mp4

```tsx
import { component$ } from "@builder.io/qwik";
import { type DocumentHead, Form, globalAction$, routeLoader$, z, zod$ } from "@builder.io/qwik-city";

interface ListItem {
  text: string;
}

export const list: ListItem[] = [];

export const useListLoader = routeLoader$(() => {
  return list;
});

export const useAddToListAction = globalAction$(
  (item) => {
    list.push(item);
    return {
      success: true
    };
  },
  zod$({
    text: z.string()
  })
);

export default component$(() => {
  const list = useListLoader();
  const action = useAddToListAction();

  return (
    <>
      <h1>Form Action TODO list</h1>
      <span>{`${action.isRunning}`}</span>{/*Here is the only code I added*/}
      <ul>
        {list.value.map((item) => (
          <li>{item.text}</li>
        ))}
      </ul>
      <Form action={action} spaReset>
        <input type="text" name="text" required />
        <button type="submit">Add item</button>
      </Form>
      <p>This little app works even when JavaScript is disabled.</p>
    </>
  );
});

export const head: DocumentHead = {
  title: "Qwik To-Do"
};

```

I did notice something interesting, that if the `<Form>` tag is inside a conditional render (no matter what the condition is) the error goes away.
```tsx
import { component$ } from "@builder.io/qwik";
import { type DocumentHead, Form, globalAction$, routeLoader$, z, zod$ } from "@builder.io/qwik-city";

interface ListItem {
  text: string;
}

export const list: ListItem[] = [];

export const useListLoader = routeLoader$(() => {
  return list;
});

export const useAddToListAction = globalAction$(
  (item) => {
    list.push(item);
    return {
      success: true
    };
  },
  zod$({
    text: z.string()
  })
);

export default component$(() => {
  const list = useListLoader();
  const action = useAddToListAction();

  return (
    <>
      <h1>Form Action TODO list</h1>
      <span>{`${action.isRunning}`}</span>{/*Here is the original code I added*/}
      <ul>
        {list.value.map((item) => (
          <li>{item.text}</li>
        ))}
      </ul>
      {true && // I added a conditional rendering here and the error goes away
        <Form action={action} spaReset>
          <input type="text" name="text" required />
          <button type="submit">Add item</button>
        </Form>
      }
      <p>This little app works even when JavaScript is disabled.</p>
    </>
  );
});

export const head: DocumentHead = {
  title: "Qwik To-Do"
};

```

**Comment by manucorporat** [comment]
I can reproduce and working on a fix!

**Comment by jihoon416** [comment]
Thank you for the fix! I can confirm my issue has been fixed when using version `0.22.0-dev20230314185952`

**Comment by rolanday** [comment]
+1 thank you!
